### PR TITLE
[macOS] Fix "on top" not set on init, and resetting on window update.

### PR DESCRIPTION
**Pull request by bruvzg**
Similar to #41285 for 3.2 branch.

### CODE REVIEW

File: platform/osx/os_osx.h [hunk 1]
```diff
@@ -127,6 +127,7 @@ class OS_OSX : public OS_Unix {
 	bool zoomed;
 	bool resizable;
 	bool window_focused;
+	bool on_top;
```
Comment (akien-mga) [inline platform/osx/os_osx.h:130]: Maybe it should be initialized to `false`?

Comment (bruvzg) [reply platform/osx/os_osx.h:130]: It's initialized in `OS_OSX::initialize`.

Comment (akien-mga) [reply platform/osx/os_osx.h:130]: Ah indeed, I was fooled by GitHub which wrongly assigns it as part of a static method:
Screenshot_20200815_170605

Comment (bruvzg) [reply platform/osx/os_osx.h:130]: Added init to false to the constructor as well.

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**